### PR TITLE
Add Quad9 DOH provider

### LINKED ISSUE #6636: Add Quad9 to DoH providers list
### Describe your suggested feature

Privacy focused DNS recursive
https://www.quad9.net/

### Other details

_No response_

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with an extension, I should be opening an issue in the [extensions repository](https://github.com/tachiyomiorg/tachiyomi-extensions/issues/new/choose).
- [X] I have updated the app to version **[0.13.1](https://github.com/tachiyomiorg/tachiyomi/releases/latest)**.
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by OncePunchedMan**
Closes #6636

### CODE REVIEW

File: app/src/main/java/eu/kanade/tachiyomi/network/DohProviders.kt [hunk 1]
```diff
@@ -52,3 +55,15 @@ fun OkHttpClient.Builder.dohAdGuard() = dns(
...
 )
+
+fun OkHttpClient.Builder.dohQuad9() = dns(
+    DnsOverHttps.Builder().client(build())
+        .url("https://dns.quad9.net/dns-query".toHttpUrl())
+        .bootstrapDnsHosts(
+            InetAddress.getByName("9.9.9.9"),
+            InetAddress.getByName("149.112.112.112"),
+            InetAddress.getByName("2620:fe::fe"),
+            InetAddress.getByName("2620:fe::9"),
```
Comment (arkon) [inline app/src/main/java/eu/kanade/tachiyomi/network/DohProviders.kt:66]: What's with the mix of base16/base10 numbers?

Comment (KokaKiwi) [reply app/src/main/java/eu/kanade/tachiyomi/network/DohProviders.kt:66]: IPv4/6 mix i suppose?

Comment (OncePunchedMan) [reply app/src/main/java/eu/kanade/tachiyomi/network/DohProviders.kt:66]: i got these directly from their website

Comment (arkon) [reply app/src/main/java/eu/kanade/tachiyomi/network/DohProviders.kt:66]: I guess it doesn't matter, it's just annoying.

Comment (OncePunchedMan) [reply app/src/main/java/eu/kanade/tachiyomi/network/DohProviders.kt:66]: true

File: app/src/main/java/eu/kanade/tachiyomi/network/NetworkHelper.kt [hunk 2]
```diff
@@ -41,6 +41,7 @@ class NetworkHelper(context: Context) {
                 PREF_DOH_CLOUDFLARE -> builder.dohCloudflare()
                 PREF_DOH_GOOGLE -> builder.dohGoogle()
                 PREF_DOH_ADGUARD -> builder.dohAdGuard()
+                PREF_DOH_QUAD9 -> builder.dohAdGuard()
```
[the remark this answers is not in the document]
Comment (OncePunchedMan) [reply app/src/main/java/eu/kanade/tachiyomi/network/NetworkHelper.kt:44]: damn oops

**Pull request merged**